crates/edr_napi/src/trace/exit.rs

```diff
@@ -0,0 +1,40 @@
+use napi::{
+    bindgen_prelude::FromNapiValue,
+    sys::{napi_env__, napi_value__},
+};
+use napi_derive::napi;
+
+#[napi]
+pub struct Exit(pub(crate) edr_solidity::exit::ExitCode);
+
+impl FromNapiValue for Exit {
+    unsafe fn from_napi_value(
```

Comment:
I wasn't sure whether I should have `#[napi] pub enum ExitCode` in edr_napi since that's also directly used in the `edr_solidity::message_trace` module; I figured that it's okay for now to have the plain Rust enum and directly implement the N-API conversion on the JS `Exit` class (compatible with the existing Hardhat interface).

Reply:
See my other comment, but I'd prefer using `#[napi] pub enum ExitCode` here to avoid the need for `unsafe` code.

Bugs in the N-API layer are particularly hard to debug, so if we can rely on auto-generated code, that's preferred.

Reply:
I followed this and have a dedicated `enum ExitCode` on the edr_solidity side with the Hardhat-specific `#[napi] pub enum ExitCode` in the edr_napi side, which does the relevant conversion.